velox/connectors/tpch/tests/TpchConnectorTest.cpp

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/tpch/TpchConnector.h"
+#include "gtest/gtest.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+
+namespace {
+
+using namespace facebook::velox;
+using namespace facebook::velox::connector::tpch;
+
+using facebook::velox::exec::test::PlanBuilder;
+using facebook::velox::tpch::Table;
+
+class TpchConnectorTest : public exec::test::OperatorTestBase {
+ public:
+  const std::string kTpchConnectorId = "test-tpch";
+
+  void SetUp() override {
+    OperatorTestBase::SetUp();
+    auto tpchConnector =
+        connector::getConnectorFactory(
+            connector::tpch::TpchConnectorFactory::kTpchConnectorName)
+            ->newConnector(kTpchConnectorId, nullptr);
+    connector::registerConnector(tpchConnector);
+  }
+
+  void TearDown() override {
+    connector::unregisterConnector(kTpchConnectorId);
+    OperatorTestBase::TearDown();
+  }
+
+  exec::Split getTpchSplit(size_t scaleFactor = 1) const {
+    return exec::Split(
+        std::make_shared<TpchConnectorSplit>(kTpchConnectorId, scaleFactor));
+  }
+
+  // Helper function to create 1:1 assignments maps based on output type.
+  auto defaultAssignments(const RowTypePtr& outputType) const {
+    std::unordered_map<std::string, std::shared_ptr<connector::ColumnHandle>>
+        assignmentsMap;
+
+    for (const auto& columnName : outputType->names()) {
+      assignmentsMap.emplace(
+          columnName, std::make_shared<TpchColumnHandle>(columnName));
+    }
+    return assignmentsMap;
+  }
+};
+
+// Simple scan of first 5 rows of "nation".
+TEST_F(TpchConnectorTest, simple) {
+  auto outputType =
+      ROW({"n_nationkey", "n_name", "n_regionkey", "n_comment"},
```

Comment:
I had the same thought while developing this. I had to add these structures here to follow the current PlanBuilder.tableScan() API, since it expects these parameters, but I guess we could also extend it and provide a tableScan() overload for TPCH that does this type resolution automatically, and only takes the column names. Thoughts?

Cc: @mbasmanova who has more context on PlanBuilder.

Reply:
It would be nice to add TPC-H specific tableScan methods that require minimum amount of information.

Reply:
I'll add something but this will require me to create the underlying api first. Is it to leave as-is for now and clean up on a follow PR?

Reply:
Sure. Thanks ! That would definitely simplify usage.